### ISSUE TITLE
Cannot add the integration, problems to choose the USB-path:    /dev/ttyUSB0     =  usb-port Odroid, problems to connect

### ISSUE THREAD

**Issue by Vermond1** [plugwise/plugwise_usb-beta#204]
### Describe the bug.

I am not an experienced HA user, but I tried to add my Plugwise Source Stick to HA. following the Github repository.

I cannot add the integration of Plugwise USB beta, problems to choose the USB-path:    (default was: /dev/ttyUSB0     )

Normally my Plugwise Source Stick is in COM3 of my windows11 computer, but that is not accepted. 

I placed the stick in the minicomputer in my windows network with HA, then the connection problems ocurred. Communication failures.

     =  usb-port Odroid minicomputer, problems to connect--> disconnect--> connect etc...try to connect.....

The result is that no entity's are found.

What do I do wrong?
 

### What version of Plugwise Beta are you using?

0.40.7

### What version of Home Assistant Core are you using?

2024.11.3

### What type of installation are you running?

Home Assistant OS

### How did you install plugwise-beta?

Manually installed `custom_component`

### What kind of Plugwise device is having issues?

USB: Stick

### What firmware version is your Plugwise product at?

Latest Source software

**Comment by bouwew** [comment]
How does your HA system look like?
Odroid running Linux? How have you installed HA on the Odroid?

**Comment by Vermond1** [comment]
Goedenavond Bouwe, 

 �

Ik heb me laten adviseren, een ervaren bekende HA gebruiker heeft de Odroid geleverd, met daarop de software geinstalleerd, incl EspHome, voor een Deye omvormer.

Ikzelf probeer me een beetje wegwijs te maken in HA, ik heb inmiddels wel de Plugwise Source circles in de lucht.(3 stuks) naast het Plugwise zone systeem.

 �

Gr Jan

 �

**Comment by bouwew** [comment]
Hoi Jan,

Als het goed is dan detecteert de Plugwise USB beta integratie het pad naar de Stick zelf.

Ik zie bijvoorbeeld dit:
![image](https://github.com/user-attachments/assets/86c666c1-f52b-45f7-86c5-44f07bee60fc)

Wat zie jij op jouw beeldscherm?

**Comment by Vermond1** [comment]
Inmiddels is het met een 2e stick die ik nog had en met de bijbehorende geresette circle+ gelukt,

Ik heb deze stick eerst met een 2e Source versie op een andere computer met de circle+ laten koppelen. (later nog 2 circles extra)

Je moet immers een werkend zigby netwerk hebben voor je verder gaat met de HA koppeling.

Van belang is natuurlijk ook dat de Odroid met HA waarin de stick zit ook binnen 5 meter van de circle+ aanwezig is. 

Bij de eerste poging was dat niet het geval. 

Thans is het wel gelukt. 

 �

Klopt het dat er per circle maar 3 entiteiten beschikbaar zijn?

 �



 �

 �

**Comment by bouwew** [comment]
Ja, je ziet er idd maar 3, de anderen zijn verborgen, die moet je handmatig aan zetten / zichtbaar maken.

**Comment by Vermond1** [comment]
Ik heb ook een paar Scans en Stealth’s van Plugwise in het netwerk. Kunnen die ook “bruikbaar” worden gemaakt via deze repository?

 �

**Comment by Vermond1** [comment]
Goedemiddag Bouwe, 

 �

Wat betreft de plugwise-usb intergratie, versie 0.40.7; 

ik gebruik HA op een Synology Nas ds220. Heb hierop de plugwise-usb integratie werkend, echter niet helemaal foutloos. Ik heb toen in de logfiles de volgende mededeling gezien. 

Met als verzoek om de author of the reposition te verwittigen.

 �

Misschien kun je er wat mee.

 �

Groet

Jan

 �

“Logger: plugwise_usb

Bron: /usr/local/lib/python3.13/site-packages/plugwise_usb/__init__.py:680

Eerst voorgekomen: 12:43:09 (36 gebeurtenissen)

Laatst gelogd: 13:08:31

 �

Error while executing callback : Detected that custom integration 'plugwise_usb' calls hass.async_create_task from a thread other than the event loop, which may cause Home Assistant to crash or data to corrupt. For more information, see https://developers.home-assistant.io/docs/asyncio_thread_safety/#hassasync_create_task at custom_components/plugwise_usb/binary_sensor.py, line 84: hass.async_create_task(async_add_binary_sensors(mac)). Please report it to the author of the 'plugwise_usb' custom integration

Error while executing callback : Detected that custom integration 'plugwise_usb' calls hass.async_create_task from a thread other than the event loop, which may cause Home Assistant to crash or data to corrupt. For more information, see https://developers.home-assistant.io/docs/asyncio_thread_safety/#hassasync_create_task at custom_components/plugwise_usb/sensor.py, line 44: hass.async_create_task(async_add_sensors(mac)). Please report it to the author of the 'plugwise_usb' custom integration

Error while executing callback : Detected that custom integration 'plugwise_usb' calls hass.async_create_task from a thread other than the event loop, which may cause Home Assistant to crash or data to corrupt. For more information, see https://developers.home-assistant.io/docs/asyncio_thread_safety/#hassasync_create_task at custom_components/plugwise_usb/switch.py, line 42: hass.async_create_task(async_add_switches(mac)). Please report it to the author of the 'plugwise_usb' custom integration”

 �

**Comment by bouwew** [comment]
@Vermond1 Ik wil voorstellen dat je upgrade naar versie v0.50.0a12, deze versie zou ook de Scan moeten ondersteunen.

**Comment by Vermond1** [comment]
Komt hierover een update  melding in HA?  Momenteel kan ik de Plugwise-USB-beta rep. In mijn HA niet eens meer vinden. Gisteren werkte het nog.

 �

**Comment by Vermond1** [comment]
Mijn HA draait in VM omgeving op een Synology NAS. De Plugwise stic zit in de Usb poort op de Nas. Het Plugwise netwerk werkt met 24 gevonden apparaten.. Het is erg gevoelig voor storingen, wsch doordat ik ook een Plugwise HA heb draaien, maar dit newerk  is zeer stabiel.

**Comment by bouwew** [comment]
Nee deze update wordt niet door HACS getoond wanneer je `Show beta` niet aan hebt staan.
Maar je kunt in HACS de update handmatig (Re)downloaden.
Let op, v0.50.0a13 is de versie die je kunt proberen.
